### PR TITLE
Fix experiment loading for SPRL

### PR DESCRIPTION
**Pull request by miterion**

### CODE REVIEW

File: Pyrado/pyrado/algorithms/utils.py [hunk 1]
```diff
@@ -269,3 +269,10 @@ def reset_rollouts(self) -> None:
     def __getattr__(self, name: str) -> Any:
         # Forward to the wrapped sampler
         return getattr(self.wrapped_sampler, name)
+
+    # Required so that pickle does not end in an endless recursion
+    def __getstate__(self):
```
Comment (fdamken) [inline Pyrado/pyrado/algorithms/utils.py:274]: Add a docstring here.

File: Pyrado/pyrado/algorithms/utils.py [hunk 2]
```diff
@@ -269,3 +269,10 @@ def reset_rollouts(self) -> None:
...
+        return vars(self)
+
+    def __setstate__(self, state):
```
Comment (fdamken) [inline Pyrado/pyrado/algorithms/utils.py:277]: Add a docstring here.

**Review by fdamken — DISMISSED** [review]

File: Pyrado/pyrado/algorithms/utils.py (hunk 1 above)
Comment (famura) [reply Pyrado/pyrado/algorithms/utils.py:274]: could move the comment above in the doc string

**Review by famura — CHANGES_REQUESTED** [review]

File: Pyrado/pyrado/algorithms/utils.py (hunk 1 above)
Comment (miterion) [reply Pyrado/pyrado/algorithms/utils.py:273]: Done

File: Pyrado/pyrado/algorithms/utils.py (hunk 2 above)
Comment (miterion) [reply Pyrado/pyrado/algorithms/utils.py:278]: Done

File: Pyrado/pyrado/algorithms/utils.py [hunk 3]
```diff
@@ -269,3 +269,13 @@ def reset_rollouts(self) -> None:
     def __getattr__(self, name: str) -> Any:
         # Forward to the wrapped sampler
         return getattr(self.wrapped_sampler, name)
+
+    def __getstate__(self):
+        """ Required so that pickle does not end in an endless recursion
+        """
```
Comment (fdamken) [inline Pyrado/pyrado/algorithms/utils.py:275]: Please move the three quotes to the end of line 274, add a period after the text and remove the surrounding spaces (Black formatting).

File: Pyrado/pyrado/algorithms/utils.py [hunk 4]
```diff
@@ -269,3 +269,13 @@ def reset_rollouts(self) -> None:
...
+        return vars(self)
+
+    def __setstate__(self, state):
+        """ Required so that pickle does not end in an endless recursion
+        """
```
Comment (fdamken) [inline Pyrado/pyrado/algorithms/utils.py:280]: Please move the three quotes to the end of line 274, add a period after the text and remove the surrounding spaces (Black formatting).

**Review by fdamken — APPROVED** [review]

**Review by fdamken — APPROVED** [review]

File: Pyrado/pyrado/algorithms/utils.py [hunk 5]
```diff
@@ -269,3 +269,11 @@ def reset_rollouts(self) -> None:
...
         # Forward to the wrapped sampler
         return getattr(self.wrapped_sampler, name)
+
+    def __getstate__(self):
+        """Required so that pickle does not end in an endless recursion"""
+        return vars(self)
+
+    def __setstate__(self, state):
+        """Required so that pickle does not end in an endless recursion"""
+        vars(self).update(state)
```
Comment (famura) [inline Pyrado/pyrado/algorithms/utils.py:279]: is this the same as `self.__dict__["foo"] = bar`?

Comment (miterion) [reply Pyrado/pyrado/algorithms/utils.py:279]: More like `self.__dict__.update(state)` but if state is defined as follows its equivalent:
```python
state = {'foo' : bar }
```

Comment (famura) [reply Pyrado/pyrado/algorithms/utils.py:279]: ok, thanks

**Pull request merged**